rskj-core/src/test/java/co/rsk/peg/BridgeSerializationUtilsTest.java

```diff
@@ -54,28 +70,135 @@
 import org.hamcrest.MatcherAssert;
 import org.junit.jupiter.api.Assertions;
 import org.junit.jupiter.api.Test;
-
 import java.math.BigInteger;
 import java.nio.charset.StandardCharsets;
 import java.time.Instant;
 import java.util.*;
 import java.util.function.Function;
 import java.util.stream.Collectors;
 
-import static org.hamcrest.CoreMatchers.is;
-import static org.hamcrest.CoreMatchers.nullValue;
-import static org.hamcrest.Matchers.*;
-import static org.junit.jupiter.api.Assertions.assertArrayEquals;
-import static org.junit.jupiter.api.Assertions.assertEquals;
-import static org.junit.jupiter.api.Assertions.assertNotEquals;
-import static org.junit.jupiter.api.Assertions.assertNull;
-import static org.junit.jupiter.api.Assertions.assertThrows;
-import static org.junit.jupiter.api.Assertions.fail;
-import static org.mockito.Mockito.mock;
-import static org.mockito.Mockito.when;
-
 class BridgeSerializationUtilsTest {
 
+    private static final NetworkParameters NETWORK_PARAMETERS = NetworkParameters.fromID(NetworkParameters.ID_MAINNET);
+    private static final Address ADDRESS = Address.fromBase58(NETWORK_PARAMETERS, "16zJJhTZWB1txoisGjEmhtHQam4sikpTd2");
+    private static final Address OTHER_ADDRESS = Address.fromBase58(NETWORK_PARAMETERS, "3NWYFKUegmqAjoW8aMhSLNVW2jy4MiA2eS");
+
+    @Test
+    void serializeAndDeserializeSingleEntry_whenValidData_shouldReturnEqualResults() {
+        // Arrange
+        BtcTransaction prevTx = new BtcTransaction(NETWORK_PARAMETERS);
+        prevTx.addOutput(Coin.FIFTY_COINS, ADDRESS);
+
+        Keccak256 key = createHash3(1);
+        BtcTransaction tx = new BtcTransaction(NETWORK_PARAMETERS);
+        tx.addInput(prevTx.getOutput(0));
+        tx.addOutput(Coin.COIN, OTHER_ADDRESS);
+
+        // Act
+        byte[] serializedEntry = 
+            BridgeSerializationUtils.serializeRskTxWaitingForSignatures(new AbstractMap.SimpleEntry<>(key, tx));
+        Map.Entry<Keccak256, BtcTransaction> deserializedEntry =
+            BridgeSerializationUtils.deserializeRskTxWaitingForSignatures(serializedEntry, NETWORK_PARAMETERS, false);
+
+        // Assert
+        assertNotNull(serializedEntry);
+        assertTrue(serializedEntry.length > 0);
+
+        assertNotNull(deserializedEntry);
+        assertEquals(key, deserializedEntry.getKey());
+        assertArrayEquals(tx.bitcoinSerialize(), deserializedEntry.getValue().bitcoinSerialize());
+    }
+
+    @Test
+    void deserializeSingleEntry_whenNullData_shouldReturnEmptyResult() {
+        // Arrange
+        byte[] data = null;
+        
+        // Act
+        Map.Entry<Keccak256, BtcTransaction> result =
+            BridgeSerializationUtils.deserializeRskTxWaitingForSignatures(data, NETWORK_PARAMETERS, false);
+        
+        // Assert
+        assertNotNull(result);
+        assertNull(result.getKey());
+        assertNull(result.getValue());
+    }
+
+    @Test
+    void deserializeSingleEntry_whenEmptyData_shouldReturnEmptyResult() {
```

Comment:
no big deal, but wdyt about parameterizing these two tests?

Reply:
Sure we can optimize a bit here :)